domains/certificates/Query.API/API.IntegrationTests/Repositories/SynchronizationMigrationTests.cs

```diff
@@ -0,0 +1,137 @@
+using API.DataSyncSyncer.Persistence;
+using API.IntegrationTests.Helpers;
+using API.IntegrationTests.Testcontainers;
+using FluentAssertions;
+using Marten;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using API.DataSyncSyncer;
+using Xunit;
+
+namespace API.IntegrationTests.Repositories;
+
+public class SynchronizationMigrationTests : IClassFixture<MartenDbContainer>, IAsyncLifetime
+{
+    private readonly IDocumentStore store;
+
+    public SynchronizationMigrationTests(MartenDbContainer dbContainer)
+        => store = DocumentStore.For(opts => opts.Connection(dbContainer.ConnectionString));
+
+    [Fact]
+    public async Task migrates_one_metering_point_with_positons()
+    {
+        var gsrn = GsrnHelper.GenerateRandom();
+        var sync1 = new SyncPosition(Guid.NewGuid(), gsrn, 42);
+        var sync2 = new SyncPosition(Guid.NewGuid(), gsrn, 43);
+        var sync3 = new SyncPosition(Guid.NewGuid(), gsrn, 44);
+
+        await SaveSyncPositions(sync1, sync2, sync3);
+
+        await store.MigrateSynchronizationPosition(CancellationToken.None);
+
+        await using var session = store.QuerySession();
+
+        var syncPositions = await session.Query<SyncPosition>().ToListAsync();
+        syncPositions.Should().HaveCount(0);
+
+        var synchronizationPositions = await session.Query<SynchronizationPosition>().ToListAsync();
+        synchronizationPositions.Should().HaveCount(1);
+        synchronizationPositions.Single().Should().BeEquivalentTo(new SynchronizationPosition { GSRN = gsrn, SyncedTo = 44 });
+    }
+
+    [Fact]
+    public async Task can_be_called_multiple_times()
+    {
+        var gsrn = GsrnHelper.GenerateRandom();
+        var sync1 = new SyncPosition(Guid.NewGuid(), gsrn, 42);
+
+        await SaveSyncPositions(sync1);
+
+        // Call multiple times
```

Comment:
instead of a comment i would suggest to extract it to a method

Reply:
No. I will remove the comment